schema/CVE_Record_Format.json

```diff
@@ -500,6 +500,93 @@
             "required": ["orgId"],
             "additionalProperties": false
         },
+        "applicability": {
+            "type": "array",
+            "items": {
+                "$ref": "#/definitions/applicabilityElement"
+            }
+        },
+        "applicabilityElement": {
+            "description": "Affected products defined using one or more software identifiers and applicability parameters. An operator property allows AND or OR logic between identifiers or combinations thereof. The negate and vulnerable Boolean properties allow applicability combinations to be inverted and/or defined as vulnerable or not. Multiple version fields are provided for capturing ranges of products when defining a Match String Range. NOTE: When defining an applicability block, it is recommended that it align with (as much as possible) the product data provided within the affected block.",
+            "properties": {
+                "operator": {
+                    "type": "string",
+                    "enum": ["AND", "OR"]
+                },
+                "negate": {
+                    "type": "boolean"
+                },
+                "nodes": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/node"
+                    }
+                }
+            }
+        },
+        "node": {
+            "description": "Defines a configuration node in an applicability statement.",
+            "properties": {
+                "operator": {
+                    "type": "string",
+                    "enum": ["AND", "OR"]
+                },
+                "negate": {
+                    "type": "boolean"
+                },
+                "cpeMatch": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/cpe_match"
+                    }
+                },
+                "purlMatch": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/purl_match"
+                    }
+                }
+            },
+            "allOf": [
+                { "required": ["operator"] },
+                {
+                    "anyOf": [
+                        { "required": ["cpeMatch"] },
+                        { "required": ["purlMatch"] }
+                    ]
+                }
+            ]
+        },
+        "purl_match": {
+            "description": "purl match string or range",
+            "type": "object",
+            "properties": {
+                "vulnerable": {
+                    "type": "boolean"
+                },
+                "criteria": {
+                    "description": "Placeholder until we find a formal purl schema",
```

Comment:
Can you elaborate?

Reply:
Right now a purl is just defined as the URI type, meaning it's a URI but that we don't constrain it in this schema itself to be a purl. It would be up to the CVE Services application to apply additional validation to ensure URLs provided are package URLs, and not other types of URLs. Alternatively, we could try to incorporate a regular expression for purls, but I'm not aware of a readily-available option for this and am hesitant to write my own lest I get it wrong. The purl spec is sufficiently complicated that I consider writing its grammar as a regular expression non-trivial.